src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -212,7 +212,7 @@ public bool DeployedOnAllNodes
         private string _lastPendingStatus;
         public MultipleUseFlag ForceReplace = new MultipleUseFlag();
 
-        protected readonly bool HandleAllDocs;
```

Comment:
Cannot we leave `readonly`? I'd prefer to set it just once in the ctor as it is now. We can pass referenced collections to base ctor and then assign it properly

Reply:
My problem is that now we need to assign it in few places while current implementation sets it just once.

Reply:
done